overlord/hookstate/ctlcmd/set.go

```diff
@@ -132,18 +156,23 @@ func (s *setCommand) setInterfaceSetting(context *hookstate.Context, plugOrSlot
 
 	var which string
 	if hookType == preparePlugHook {
-		which = "plug-attrs"
+		which = "plug"
 	} else {
-		which = "slot-attrs"
+		which = "slot"
 	}
 
-	st := context.State()
-	st.Lock()
-	defer st.Unlock()
+	context.Lock()
+	defer context.Unlock()
 
-	attributes := make(map[string]interface{})
-	if err := attrsTask.Get(which, &attributes); err != nil {
-		return fmt.Errorf(i18n.G("internal error: cannot get %s from appropriate task"), which)
+	staticAttrs := make(map[string]interface{})
+	dynamicAttrs := make(map[string]interface{})
+	if err = attrsTask.Get(fmt.Sprintf("%s-static", which), &staticAttrs); err != nil {
+		return fmt.Errorf(i18n.G("internal error: cannot get %s from appropriate task, %s"), which, err)
+	}
+
+	dynKey := fmt.Sprintf("%s-dynamic", which)
```

Comment:
Nitpick: can you roll the `dynKey` variable into the `Get` call below as you did above?

Reply:
I need it again a few lines below, that's why it goes into dynKey.